analyzers/tests/SonarAnalyzer.TestFramework/Common/SourceGeneratorProvider.cs

```diff
@@ -26,13 +26,31 @@ namespace SonarAnalyzer.TestFramework.Common;
 public static class SourceGeneratorProvider
 {
     private static readonly string RazorSourceGeneratorPath =
-        Path.Combine(Path.GetDirectoryName(typeof(SourceGeneratorProvider).Assembly.Location), "Dependencies", "Microsoft.CodeAnalysis.Razor.Compiler.SourceGenerators.dll");
+        Path.Combine(LatestSdkFolder(), "Sdks", "Microsoft.NET.Sdk.Razor", "source-generators", "Microsoft.CodeAnalysis.Razor.Compiler.SourceGenerators.dll");
 
     public static AnalyzerFileReference[] SourceGenerators { get; } =
     [
-        new(RazorSourceGeneratorPath, new AssemblyLoader())
+        new(CheckRazorSourceGeneratorPath(), new AssemblyLoader())
     ];
 
+    public static string CheckRazorSourceGeneratorPath() =>
+        File.Exists(RazorSourceGeneratorPath) ? RazorSourceGeneratorPath : throw new FileNotFoundException($"Razor sourcegenerator not found: {RazorSourceGeneratorPath}");
```

Comment:
Defensive check, not easily testable

Reply:
I would call this overengineering. 
This produces
```
   System.TypeInitializationException: The type initializer for 'SonarAnalyzer.TestFramework.Common.SourceGeneratorProvider' threw an exception. ---> System.IO.FileNotFoundException: Razor sourcegenerator not found: C:\Program Files\dotnet\sdk\8.0.204\Sdks\Microsoft.NET.Sdk.Razor\source-generators\Microsoft.CodeAnalysis.Razor.Compiler.SourceGeneratorsMissing.dll
```
while if we let the loader load nonexistent file, we'll get practically exactly the same
```
System.IO.FileNotFoundException: Could not load file or assembly 'C:\Program Files\dotnet\sdk\8.0.204\Sdks\Microsoft.NET.Sdk.Razor\source-generators\Microsoft.CodeAnalysis.Razor.Compiler.SourceGeneratorsMissing.dll'. The system cannot find the file specified.
```

so I don't see any added value for the untestable code => it doesn't need to be there.

Reply:
It also affects only test code, so there's no risk for users at all

Reply:
It's not about the risk for users, it's about the risk for us to depend on a local file that might become hard to debug/reproduce. The code intended to reduce it as much as possible, but I see your point, and it's probably not that helpful, I'll delete this part